web_client/dialogs/editAnnotation.js

```diff
@@ -0,0 +1,95 @@
+import tinycolor from 'tinycolor2';
+
+import View from 'girder/views/View';
+
+import editAnnotation from '../templates/dialogs/editAnnotation.pug';
+import 'girder/utilities/jquery/girderModal';
+
+/**
+ * Create a modal dialog with fields to edit the properties of
+ * an annotation element.
+ */
+var EditAnnotation = View.extend({
+    events: {
+        'click .h-submit': 'getData',
+        'submit form': 'getData'
+    },
+
+    render() {
+        this.$el.html(
+            editAnnotation({
+                element: this.annotationElement.toJSON()
+            })
+        ).girderModal(this);
+        this.$('.h-colorpicker').colorpicker();
+    },
+
+    /**
+     * Get all data from the form and set the attributes of the
+     * attached ElementModel (triggering a change event).
+     */
+    getData(evt) {
+        evt.preventDefault();
+
+        var data = {};
+        var label = this.$('#h-element-label').val();
+        if (label) {
+            data.label = {
+                value: label
+            };
+        }
+
+        var lineWidth = this.$('#h-element-line-width').val();
+        if (lineWidth) {
+            data.lineWidth = parseFloat(lineWidth);
```

Comment:
In general, do we need checks to make sure senseless values are rejected?  For instance, I can enter negative numbers.